agent/connect/ca/provider_vault_test.go

```diff
@@ -237,8 +238,68 @@ func TestVaultCAProvider_Configure(t *testing.T) {
 			testcase.expectedValue(t, provider)
 		})
 	}
+}
+
+// This test must not run in parallel
+func TestVaultCAProvider_ConfigureWithBadToken(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+	SkipIfVaultNotPresent(t)
+
+	testVault := NewTestVaultServer(t)
+
+	attr := &VaultTokenAttributes{
+		RootPath:         "pki-root",
+		IntermediatePath: "pki-intermediate",
+		ConsulManaged:    true,
+	}
+	token := CreateVaultTokenWithAttrs(t, testVault.client, attr)
 
-	return
+	provider := NewVaultProvider(hclog.New(&hclog.LoggerOptions{Name: "ca.vault"}))
+
+	t.Run("bad token does not leak renewal routine on Configure", func(t *testing.T) {
+		config := map[string]any{
+			"RootPKIPath":         "pki-root/",
+			"IntermediatePKIPath": "badbadbad/",
+		}
+		cfg := vaultProviderConfig(t, testVault.Addr, token, config)
+
+		err := provider.Configure(cfg)
+		require.Error(t, err)
+
+		retry.RunWith(retry.TwoSeconds(), t, func(r *retry.R) {
+			profile := pprof.Lookup("goroutine")
+			sb := strings.Builder{}
+			require.NoError(r, profile.WriteTo(&sb, 2))
```

Comment:
pprof is a runtime profile which can dump goroutines.
```
profile.WriteTo(&sb, 2)
```
The 2 here means "format it like golang stacktraces".

We are looking for this trace in particular:
```
        goroutine 138 [select]:
        github.com/hashicorp/consul/agent/connect/ca.(*VaultProvider).renewToken(0xc00077bc20, {0x28f2bd8, 0xc0008ce6e0}, 0xc0008bae00)
        	/Users/chriskim/code/consul/agent/connect/ca/provider_vault.go:240 +0x1c5
        created by github.com/hashicorp/consul/agent/connect/ca.(*VaultProvider).Configure in goroutine 135
        	/Users/chriskim/code/consul/agent/connect/ca/provider_vault.go:184 +0x67c
```

Reply:
Great job thinking through and implementing a testing solution to this! Not a straightforward, run-of-the-mill unit testing problem :+1: